lgn-provers/src/provers/v1/preprocessing/mod.rs

```diff
@@ -1,11 +1,13 @@
 use crate::provers::v1::preprocessing::prover::{StorageDatabaseProver, StorageExtractionProver};
 use crate::provers::v1::preprocessing::task::Preprocessing;
 use tracing::info;
-mod dummy_prover;
 pub mod euclid_prover;
 pub mod prover;
 pub mod task;
 
+#[cfg(feature = "dummy-prover")]
+mod dummy_prover;
```

Comment:
these fix the clippy warnings about dummy prover never being constructed and remove the need for the allow dead code.